hera_qm/ant_metrics.py

```diff
@@ -431,3 +436,155 @@ def save_antenna_metrics(self, metricsJSONFilename):
+    fn = re.findall('zen\.\d{7}\.\d{5}\..*', fname)[0]
+    return fn.split('.')[3]
+
+def generate_fullpol_file_list(files, pol_list):
+    """Generate a list of unique JDs that have all four polarizations available
+    Args:
+       files -- list of files to look for
+       pol_list -- list of polarizations to look for
+    Returns:
+       jd_list -- list of JDs where all supplied polarizations could be found
+
+    This function, when given a list of files, will look for the specified polarizations,
+    and add the JD to the returned list if all polarizations were found.
+    """
+    # initialize
+    file_list = []
+
+    for filename in files:
+        abspath = os.path.abspath(filename)
+        if abspath not in file_list:
+            # try to find the other polarizations
+            pols_exist = True
+            file_pol = get_pol(filename)
+            dirname = os.path.dirname(abspath)
+            for pol in pol_list:
+                # guard against strange directory names that might contain something that
+                # looks like a pol string
+                fn = re.sub(file_pol, pol, filename)
+                full_filename = os.path.join(dirname, fn)
+                if not os.path.exists(full_filename):
+                    warnings.warn("Could not find " + full_filename + "; skipping that JD")
+                    pols_exist = False
+                    break
+            if pols_exist:
+                # add all pols to file_list
+                for pol in pol_list:
+                    fn = re.sub(file_pol, pol, filename)
+                    full_filename = os.path.join(dirname, fn)
+                    file_list.append(full_filename)
+
+    return file_list
+
+def ant_metrics_run(files, opts):
+    """
+    Run a series of ant_metrics tests on a given set of input files.
+
+    Args:
+       files -- a list of files to run ant metrics on. Can be any of the 4 polarizations
+       opts -- an optparse OptionParser instance
+    Return:
+       None
+
+    The funciton will take in a list of files and options. It will run the
+    sereis of ant metrics tests, and produce a JSON file containing the relevant
+    information. The file list passed in need only contain one of the polarization
+    files for a given JD, and the function will look for the other polarizations
+    in the same folder. If not all four polarizations are found, a warning is
+    generated, since the code assumes all four polarizations are present.
+    """
+    # XXX: does this create a circular dependency if we only do this inside the function?
+    try:
+        from hera_cal import omni
+    except(ImportError):
+        from nose.plugins.skip import SkipTest
+        raise SkipTest('hera_cal.omni not detected. It must be installed to calculate array info')
+
+    # define polarizations to look for
+    # XXX: This will have to change if file naming convention changes from 'xx', 'xy', etc.
+    pol_list = ['xx', 'xy', 'yx', 'yy']
```

Comment:
My thinking behind this is that the helper function `generate_fullpol_file_list` needs to know what a "complete set" of polarized visibilities is. I suppose we could have the options include a `-p` option like `omni_run`, and if it's not specified, we default to this pol_list.